packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -54,21 +55,21 @@ export async function tryHotswapDeployment(
   // create a new SDK using the CLI credentials, because the default one will not work for new-style synthesis -
   // it assumes the bootstrap deploy Role, which doesn't have permissions to update Lambda functions
   const sdk = (await sdkProvider.forEnvironment(resolvedEnv, Mode.ForWriting)).sdk;
-  // The current resources of the Stack.
-  // We need them to figure out the physical name of a resource in case it wasn't specified by the user.
-  // We fetch it lazily, to save a service call, in case all hotswapped resources have their physical names set.
-  const listStackResources = new LazyListStackResources(sdk, stackArtifact.stackName);
```

Comment:
this has been moved elsewhere, could you explain why? I don't see why this is necessary.